### PR TITLE
Fga 13

### PR DESCRIPTION
**Pull request by MasterKiller1239**
Pagination added

### CODE REVIEW

File: gitapp/lib/users_list/home_page_view.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  HomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  HomePageState createState() => HomePageState();
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  final UsersPresenter users = UsersPresenter();
```
Comment (krecikzgr) [inline gitapp/lib/users_list/home_page_view.dart]: It should be named Presenter not the users.

File: gitapp/lib/users_list/home_page_view.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      await users.updateSearchedListFromApi(text);
+      setState(() {
+        searchBar.textController.clear();
+        searching = false;
+      });
+    } else
+      users.searchedList.clear();
+  }
+
+  Widget chooseWidget() {
```
Comment (krecikzgr) [inline gitapp/lib/users_list/home_page_view.dart]: What does it mean `chooseWidget` ? Choose widget from what? The name of this method is not providing us the proper context and someone which doest not know the code won't understand this.

Comment (krecikzgr) [reply gitapp/lib/users_list/home_page_view.dart]: Also this logic should be moved to presenter. This is logic connected with connectionStatus. Not in the view

File: gitapp/lib/users_list/home_page_view.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+class HomePageState extends State<HomePage> with TickerProviderStateMixin {
+  final UsersPresenter users = UsersPresenter();
+  late Searchbar searchBar = new Searchbar(getUsers);
+  bool connectionStatus = true;
+  bool searching = false;
+
+  Future getUsers(String text) async {
+    connectionStatus = await Connection.instance.checkConnection();
```
Comment (krecikzgr) [inline gitapp/lib/users_list/home_page_view.dart]: Checking of the connection should be made in presenter. The presenter os the source of our logic, and in there we should perform all checks, and clearing of the arrays and so on.

File: gitapp/lib/users_list/users_presenter.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+class UsersPresenter {
+  List<User> userList = List.empty(growable: true);
+  List<User> searchedList = List.empty(growable: true);
+  int currentPage = 1;
+  String currentUsername = "";
+
+  UsersPresenter();
+
+  Future<void> updateSearchedListFromApi(String username) async {
```
Comment (krecikzgr) [inline gitapp/lib/users_list/users_presenter.dart]: There is some problem or misunderstanding.

Users_presenter should be responsible for the logic. Not only for data searching.

**Review by krecikzgr — APPROVED** [review]

File: gitapp/lib/users_list/home_page_view.dart (hunk 1 above)
Comment (MasterKiller1239) [reply gitapp/lib/users_list/home_page_view.dart]: Solved

File: gitapp/lib/users_list/users_presenter.dart (hunk 4 above)
Comment (MasterKiller1239) [reply gitapp/lib/users_list/users_presenter.dart]: Users_presenter now sends state to the view which tells it what to show on screen

File: gitapp/lib/users_list/home_page_view.dart (hunk 3 above)
Comment (MasterKiller1239) [reply gitapp/lib/users_list/home_page_view.dart]: Solved

File: gitapp/lib/users_list/home_page_view.dart (hunk 2 above)
Comment (MasterKiller1239) [reply gitapp/lib/users_list/home_page_view.dart]: > Also this logic should be moved to presenter. This is logic connected with connectionStatus. Not in the view

Connection is now in presenter.

Comment (MasterKiller1239) [reply gitapp/lib/users_list/home_page_view.dart]: > What does it mean `chooseWidget` ? Choose widget from what? The name of this method is not providing us the proper context and someone which doest not know the code won't understand this.

name changed to chooseWidgetBasedOnState

**Pull request merged**